File: src/main/java/com/ziyad/securenotetakingapplication/service/FolderService.java
package com.ziyad.securenotetakingapplication.service;

import com.ziyad.securenotetakingapplication.model.User;
import com.ziyad.securenotetakingapplication.payload.FolderDTO;
import com.ziyad.securenotetakingapplication.payload.FolderResponse;
import org.springframework.stereotype.Service;

@Service
public interface FolderService {
    FolderDTO createFolder(User user, FolderDTO folderDTO);

    FolderResponse getAllFolders(User user, Integer pageNumber, Integer pageSize, String sortBy, String sortOrder);

    FolderDTO updateFolder(User byUsername, FolderDTO folderDTO, Long folderId);

    FolderDTO deleteFolder(User user, Long folderId);

}

File: secure-note-taking-application-frontend/src/services/folderService.js
import axios from 'axios';
import Cookies from 'js-cookie';

const API_URL = process.env.REACT_APP_API_URL;

const getAuthHeader = () => {
    const token = Cookies.get('SecureNoteTakingApplication');
    if (token) {
        return { Authorization: token };
    } else {
        return {};
    }
};

const getAllFolders = (pageNumber, pageSize, sortBy, sortOrder) => {
    const token = Cookies.get('SecureNoteTakingApplication');
    return axios.get(`${API_URL}/folder/get`, {
        params: { pageNumber, pageSize, sortBy, sortOrder },
        headers: { 
            Authorization: token 
        },
        withCredentials: true
    });
};

const getFolder = (folderId) => {
    return axios.get(`${API_URL}/folder/${folderId}`, { headers: getAuthHeader(), withCredentials: true });
};

const createFolder = (folderName) => {
    return axios.post(`${API_URL}/folder/create`, { folderName }, { headers: getAuthHeader(), withCredentials: true });
};

const updateFolder = (folderId, folderName) => {
    return axios.put(`${API_URL}/folder/update/${folderId}`, { folderName }, { headers: getAuthHeader(), withCredentials: true });
};

const deleteFolder = (folderId) => {
    return axios.delete(`${API_URL}/folder/delete/${folderId}`, { headers: getAuthHeader(), withCredentials: true });
};

const folderService = { getAllFolders, getFolder, createFolder, updateFolder, deleteFolder };

export default folderService;